.github/workflows/ci.yml

```diff
@@ -103,3 +104,85 @@ jobs:
       uses: katyo/publish-crates@v1
       with:
           registry-token: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+  
+  build-publish-master-image:
+    needs: build
+    if: github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+            
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: |
+            ghcr.io/fuellabs/sway
+          tags: |
+            type=sha
+            type=ref,event=branch
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      
+      - name: Log in to the ghcr.io registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+    
+      - name: Build and push the image to ghcr.io
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: deployment/Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max  
+  
+  build-publish-release-image:
+    # Build & Publish Docker Image Per Fuel-Core Release
+    needs: build
```

Comment:
Can you make this depend on `publish` instead of `build`? Then you don't even need the check that the event is a release. Alternatively, append the job tasks to the end of the `publish` job.

Reply:
thanks this is a good suggestion/improvement- updated the PR.